batavia/types/Dict.js

```diff
@@ -4,305 +4,11 @@
 
 batavia.types.Dict = function() {
     function Dict(args, kwargs) {
-        Object.call(this);
-        if (args) {
-            this.update(args);
-        }
+        batavia.types.JSDict.call(this, args, kwargs);
     }
 
-    Dict.prototype = Object.create(Object.prototype);
+    Dict.prototype = Object.create(batavia.types.JSDict.prototype);
```

Comment:
Can you elaborate on why this separation into JSDict is necessary? It isn't obvious from this patch.